### PR TITLE
Remove s3 handler map in s3 proxy

### PR DESCRIPTION
**Pull request by humengyu2012**
### What changes are proposed in this pull request?

Remove s3 handler map in s3 proxy.

### Why are the changes needed?

https://github.com/Alluxio/alluxio/issues/17403

### Does this PR introduce any user facing changes?

Please list the user-facing changes introduced by your change, including
  1. change in user-facing APIs NO
  2. addition or removal of property keys NO
  3. webui NO

### CODE REVIEW

File: core/server/proxy/src/main/java/alluxio/web/ProxyWebServer.java [hunk 1]
```diff
@@ -72,13 +71,12 @@ public final class ProxyWebServer extends WebServer {
...
   private AsyncUserAccessAuditLogWriter mAsyncAuditLogWriter;
-  public static final String PROXY_S3_HANDLER_MAP = "Proxy S3 Handler Map";
-  public ConcurrentHashMap<Request, S3Handler> mS3HandlerMap = new ConcurrentHashMap<>();
+  public static final String S3_HANDLER_ATTRIBUTE = "Proxy S3 Handler Attribute";
 
   class ProxyListener implements HttpChannel.Listener {
     public void onComplete(Request request)
     {
-      S3Handler s3Hdlr = mS3HandlerMap.get(request);
+      S3Handler s3Hdlr = (S3Handler) request.getAttribute(S3_HANDLER_ATTRIBUTE);
```
Comment (lucyge2022) [inline core/server/proxy/src/main/java/alluxio/web/ProxyWebServer.java:79]: will it NPE when no attributes gets set?

Comment (humengyu2012) [reply core/server/proxy/src/main/java/alluxio/web/ProxyWebServer.java:79]: > will it NPE when no attributes gets set?

`S3Handler s3Hdlr = (S3Handler) null;` means s3Hdlr is null, like 'org.eclipse.jetty.server.Server#Server()'

**Review by lucyge2022 — APPROVED** [review]
LGTM

**Comment by HelloHorizon** [comment]
alluxio-bot, merge this please

**Comment by Jackson-Wang-7** [comment]
alluxio-bot, cherry-pick this to branch-2.10 please

**Pull request merged**